### PR TITLE
RHPAM-2307[7.26.x] 

### PR DESCRIPTION
**Pull request by lanceleverich**
Cherry-picked for 7.5.1 release, this includes fixes for:
* DROOLS-4467 - PMML with capitalized field name does not work
* RHPAM-2307 - Guided Scorecard can not be tested according to its name
Both of these fixes have been committed to the master branch

### CODE REVIEW

**Comment by lanceleverich** [comment]
Jenkins retest this

File: drools-core/src/main/java/org/drools/core/command/runtime/pmml/ApplyPmmlModelCommand.java [hunk 1]
```diff
@@ -158,13 +159,20 @@ public void setOutIdentifier(String outIdentifier) {
...
+                }
             }
         }
         String basePkgName = PmmlConstants.DEFAULT_ROOT_PACKAGE+"."+javaModelId;
         packageNames.add(basePkgName);
+        if (!javaModelId.equals(capJavaModelId)) {
+            packageNames.add(PmmlConstants.DEFAULT_ROOT_PACKAGE + "." + capJavaModelId);
+        }
         return packageNames;
     }
```
Comment (jomarko) [inline drools-core/src/main/java/org/drools/core/command/runtime/pmml/ApplyPmmlModelCommand.java:177]: To me it seems we now enforce always to have UperCase model id, so thinking if we can not simplify as:
```java
    private List<String> calculatePossiblePackageNames(String modelId, String...knownPackageNames) {
        List<String> packageNames = new ArrayList<>();
        String javaModelId = modelId.replaceAll("\\s","");
        String capJavaModelId = StringUtils.ucFirst(javaModelId);
        if (knownPackageNames != null && knownPackageNames.length > 0) {
            for (String knownPkgName: knownPackageNames) {
                packageNames.add(knownPkgName + "." + capJavaModelId);
            }
        }
        String basePkgName = PmmlConstants.DEFAULT_ROOT_PACKAGE + "." + capJavaModelId;
        packageNames.add(basePkgName);
        return packageNames;
    }
```

File: drools-scorecards/src/test/java/org/drools/scorecards/TestUtil.java [hunk 2]
```diff
@@ -51,13 +52,20 @@
...
+                }
             }
         }
         String basePkgName = DEFAULT_ROOT_PACKAGE + "." + javaModelId;
         packageNames.add(basePkgName);
+        if (!javaModelId.equals(capJavaModelId)) {
+            packageNames.add(DEFAULT_ROOT_PACKAGE + "." + capJavaModelId);
+        }
         return packageNames;
     }
```
Comment (jomarko) [inline drools-scorecards/src/test/java/org/drools/scorecards/TestUtil.java:70]: We have exactly the same method in `ApplyPmmlModelCommand` class, that is in `drools-core` module. As the `drools-scorecards` has already a dependency for `drools-core` I think we could make `calculatePossiblePackageNames` available as static util method in `drools-core` module and then we do not have to introduce `calculatePossiblePackageNames` in `drools-scorecards`.

File: drools-workbench-models/drools-workbench-models-guided-scorecard/src/test/java/org/drools/workbench/models/guided/scorecard/backend/test2/GuidedScoreCardIntegrationJavaClassesAddedToKieFileSystemTest.java [hunk 3]
```diff
@@ -239,13 +240,20 @@ public void testIncrementalCompilation() {
...
+                }
             }
         }
         String basePkgName = DEFAULT_ROOT_PACKAGE + "." + javaModelId;
         packageNames.add(basePkgName);
+        if (!javaModelId.equals(capJavaModelId)) {
+            packageNames.add(DEFAULT_ROOT_PACKAGE + "." + capJavaModelId);
+        }
         return packageNames;
     }
```
Comment (jomarko) [inline drools-workbench-models/drools-workbench-models-guided-scorecard/src/test/java/org/drools/workbench/models/guided/scorecard/backend/test2/GuidedScoreCardIntegrationJavaClassesAddedToKieFileSystemTest.java:258]: The same comment, we have already such method in `drools-core` and we have a dependency for `drools-core`

File: kie-pmml/src/main/java/org/kie/pmml/pmml_4_2/PMML4ExecutionHelper.java [hunk 4]
```diff
@@ -352,12 +352,19 @@ public synchronized PMML4Result submitRequest(PMMLRequestData request)
...
+                    packageNames.add(knownPkgName + "." + capJavaModelId);
+                }
             }
         }
         packageNames.add(DEFAULT_ROOT_PACKAGE + "." + javaModelId);
+        if (!javaModelId.equals(capJavaModelId)) {
+            packageNames.add(DEFAULT_ROOT_PACKAGE + "." + capJavaModelId);
+        }
         return packageNames;
     }
```
Comment (jomarko) [inline kie-pmml/src/main/java/org/kie/pmml/pmml_4_2/PMML4ExecutionHelper.java:369]: Same comment as above.

**Review by jomarko — COMMENTED** [review]
Some comments/questions.

File: drools-core/src/main/java/org/drools/core/command/runtime/pmml/ApplyPmmlModelCommand.java (hunk 1 above)
Comment (jomarko) [reply drools-core/src/main/java/org/drools/core/command/runtime/pmml/ApplyPmmlModelCommand.java:177]: I tried manually and it seems we can not simplify, so please ignore the comment above, however I think comments about reusability are still valid.

**Pull request merged**